external/implementations/samlconf-openam-impl/README.md

```diff
@@ -0,0 +1,58 @@
+| (Does not properly use RelayStates) | Bindings 3.5.4 | If a "RelayState" value is to accompany the SAML protocol message, it MUST be placed in an additional hidden form control named RelayState within the same form with the SAML message.
+| (Does not properly use Relay States) | Bindings 3.4.3 | If a SAML request message is accompanied by RelayState data, then the SAML responder MUST return its SAML protocol response using a binding that also supports a RelayState mechanism, and it MUST place the exact data it received with the request into the corresponding RelayState parameter in the response.
+| Does not return a SAML Response Error when given a Non-Matching destination | Core 3.2.1 | If [Destination] is present, the actual recipient MUST check that the URI reference identifies the location at which the message was received. If it does not, the request MUST be discarded.
+
+## Steps to Test OpenAM's IDP
+#### install and run OpenAM
+* Download [Apache Tomcat](https://tomcat.apache.org/download-70.cgi).
+* Download the OpenAM .war files from [OpenAM Releases](https://github.com/OpenIdentityPlatform/OpenAM/releases).
+* Rename OpenAM-\<version\>.war to OpenAM.war.
+* Unzip Apache Tomcat and move all OpenAm .war files into <tomcat>/webapps.
+* Run the following commands to start tomcat:
+  * `chmod -R 755 <tomcat>/bin`
+  * `export CATALINA_HOME=<tomcat>`
+  * `sh <tomcat>/bin/startup.sh`
+
+* Follow the steps [here](https://github.com/OpenIdentityPlatform/OpenAM/wiki/Quick-Start-Guide#basic-openam-setup) until Policy Configuration.
+
+#### Setup OpenAM as a hosted IDP
+* Once you've logged in as amAdmin, click Top Level Realm \> Configure SAMLv2 Provider \> Create Hosted Identify Provider.
+
+* Fill out the form (shown below) by picking a signing key, entering a name for the Circle of Trust (CoT) and clicking Configure.
+![Create A Hosted IdP](images/OpenAM-CreateIdP.png "Setting up an IdP")
+  * **NOTE**: "http://localhost:8080/OpenAM-14.1.13" would be the **baseURL** and "/" would be the **realmName** in the above example.
+
+#### Registering and Configuring the CTK
+* Register a new Service Provider under the same Circle of Trust (CoT) and give it the following metadata: [`samlconf-sp-metadata.xml`](../../../deployment/distribution/src/main/resources/samlconf-sp-metadata.xml)
+
+* Navigate to the SAML2 page on OpenAM, (Realm \> Applications \> SAML 2.0).
+* Click on the CoT that both the CTK and the IDP are a part of.
+* On the left under "available" there should be `https://samlhostdsa:8994`. Add it to the CoT and save.
+
+* Setup a new user for the SamlCTK to connect as, (Realm \> Subjects \> New...).
+  * ID: `ctk`
+  * Password: `gotestit`
```

Comment:
The reason I had to go away from `admin:admin` was because openAM by default forces passwords to be at least 8 characters. I can change the username to `admin` but the password would still need to be something that's 8 characters

Reply:
That way the -u flag is mandatory for OpenAM